api/views.py

```diff
@@ -108,3 +108,70 @@ def create(self, request, **kwargs):
                 )
 
         return Response(PuzzleSerializer(puzzle).data)
+
+
+class PuzzleTagViewSet(viewsets.ModelViewSet):
+    permission_classes = [IsAuthenticated]
+    serializer_class = PuzzleTagSerializer
+
+    def get_queryset(self):
+        hunt_id = self.kwargs["hunt_id"]
+        return PuzzleTag.objects.filter(hunt__id=hunt_id)
+
+    def destroy(self, request, pk=None, **kwargs):
+        puzzle = None
+        with transaction.atomic():
+            tag = self.get_object()
+            puzzle = get_object_or_404(Puzzle, pk=self.kwargs["puzzle_id"])
+            if puzzle.name == tag.name:
+                return Response(
+                    {
+                        "detail": "You cannot remove a meta's tag (%s) from itself"
+                        % tag.name
+                    },
+                    status=400,
+                )
+            if tag.is_meta:
+                meta = Puzzle.objects.get(name=tag.name, hunt=puzzle.hunt)
+                # the post m2m hook will remove tag
```

Comment:
Have we considered creating m2m hooks on the PuzzleTag object that checks `is_meta` and makes these updates accordingly? That way, we could probably just call `puzzle.tags.remove(tag)` here and `puzzle.tag.add(tag)` below and avoid having these conditionals. It seems a bit strange to use the `metas` field in some cases and `tags` field in other cases, when we're really creating/deleting tag objects under the hood.

Reply:
Yeah, that makes sense to me (this was just the existing structure copied over to this file), especially if we're always going to be going in the tag => meta direction.